File: 프로그래머스/0/120826. 특정 문자 제거하기/특정 문자 제거하기.py
def solution(my_string, letter):
    # [풀이1] replace()를 사용하여 letter를 빈 문자열로 치환
    # return my_string.replace(letter, '')

    # [풀이2] 반복문을 이용하여 letter가 아닌 문자만 추가
    answer = ''

    for m in my_string:  # 문자열의 각 문자 순회
        if m != letter:  # letter와 다른 문자만 추가
            answer += m

    return answer  # 변환된 문자열 반환

# 테스트 예제 실행
print(solution("abcdef", "f"))      # "abcde"
print(solution("BCBdbe", "B"))      # "Cdbe"

File: 프로그래머스/0/120826. 특정 문자 제거하기/특정 문자 제거하기.java
class Solution {
    public String solution(String myString, String letter) {
        String result = ""; // 결과 문자열 초기화

        // 문자열 순회하며 letter와 다른 문자만 추가
        for (char c : myString.toCharArray()) {
            if (c != letter.charAt(0)) { // letter와 다른 문자만 추가
                result += c;
            }
        }

        return result; // 최종 변환된 문자열 반환
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        // 테스트 예제 실행
        System.out.println(sol.solution("abcdef", "f")); // "abcde"
        System.out.println(sol.solution("BCBdbe", "B")); // "Cdbe"
    }
}
